Update `openOrCreate` to `open`

ayuishii opened the pull request:
Issue: https://github.com/WICG/storage-buckets/issues/26

pwnall reviewed (changes requested):
Please add a section to `Alternatives considered` that mentions the old name and explains pros (clearer description of what's happening) and cons (`open` is more recognizable across storage APIs; the behavior here matches what `open` does for many APIs)

ayuishii commented:
Thanks for the review!
I added a "Opening and creating buckets" section, and added the naming as a sub-section.
Let me know if this matches what you had in mind. 
Thanks!

pwnall reviewed (approved):
LGTM. Thank you for the revisions!
